main/middleware/apisix_user.py

```diff
@@ -33,7 +49,77 @@ def decode_apisix_headers(request, header):
         return None
 
     log.debug("decode_apisix_headers: Got %s", apisix_result)
-    return {"username": apisix_result.get("preferred_username", None)}
+    return {
+        modelKey: apisix_result[data_mapping[modelKey]]
+        for modelKey in data_mapping
+        if data_mapping[modelKey] in apisix_result
+    }
+
+
+def get_user_from_apisix_headers(request, decoded_headers, original_header):
+    """
+    Get a user based on the APISIX headers, create user/profile if needed.
+
+    Args:
+        request: Django request object
+        decoded_headers: Decoded APISIX headers
+        original_header: Original header
+
+    Returns:
+        User object
+
+    """
+
+    User = get_user_model()
+
+    if not decoded_headers:
+        return None
+
+    global_id = decoded_headers.get("global_id")
+    if not global_id:
+        log.error("No global_id found in APISIX headers")
+        return None
+
+    user, created = User.objects.update_or_create(
+        global_id=global_id,
+        defaults={
+            "username": decoded_headers.get("username", ""),
+            "email": decoded_headers.get("email", ""),
+            "first_name": decoded_headers.get("first_name", ""),
+            "last_name": decoded_headers.get("last_name", ""),
```

Comment:
We need to also update `profile.name`, because otherwise new users don't have their name show up. Does APISIX pass this?

Reply:
updated to get it from fullName, which should now be sent from the local keycloak instance